Sources/Paywalls/Events/Networking/PaywallEventsRequest.swift

```diff
@@ -16,20 +16,39 @@ import Foundation
 /// The content of a request to the events endpoints.
 struct PaywallEventsRequest {
 
-    var events: [Event]
+    var events: [AnyEncodable]
 
-    init(events: [Event]) {
+    init(events: [AnyEncodable]) {
         self.events = events
     }
 
     @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
     init(events: [PaywallStoredEvent]) {
-        self.init(events: events.map { .init(storedEvent: $0) })
+        self.init(events: events.compactMap { storedEvent in
+            switch storedEvent.feature {
+            case .paywalls:
+                guard let event = PaywallEventRequest.Event(storedEvent: storedEvent) else {
+                    return nil
+                }
+                return AnyEncodable(event)
+        })
     }
 
 }
 
-extension PaywallEventsRequest {
+protocol FeatureEvent: Encodable {
+
+    var id: String? { get }
+    var version: Int { get }
+    var appUserID: String { get }
+    var sessionID: String { get }
+
+}
+
+// This is a `struct` instead of `enum` so that
+// we can use make it conform to Encodable
+// swiftlint:disable:next convenience_type
+struct PaywallEventRequest {
 
     enum EventType: String {
```

Comment:
This used to be part of the `PaywallEventsRequest` but we now need to have different `EventType` for `PaywallEventRequest` and `CustomerCenterRequest`